meson.build

```diff
@@ -0,0 +1,76 @@
+# SPDX-License-Identifier: GPL-2.0
+
+project('cdba',
+	'c',
+	default_options: [
+            'warning_level=2',	# sets -Wextra
+            'buildtype=release',
+        ])
+
+compiler = meson.get_compiler('c')
+
+# If compiler variables are detectable, add some selected ones
+if meson.version().version_compare('>=0.43.0')
+	base_cflags = ['-Wno-unused-parameter',
+		       '-Wno-unused-result',
+		       '-Wno-missing-field-initializers',
+		       '-Wno-sign-compare',
+		       '-Wundef',
+		       '-Wnull-dereference',
+		       '-Wdouble-promotion',
+		       '-Wshadow',
+		       '-Wpointer-arith',
+		       '-Wwrite-strings',
+		       '-Wstrict-overflow=4']
+
+	if compiler.get_id() == 'gcc'
+		compiler_cflags = ['-Werror',	# Only set it on GCC
+			           '-Wformat-signedness',
+				   '-Wduplicated-cond',
+				   '-Wduplicated-branches',
+				   '-Wvla-larger-than=1',
+				   '-Walloc-zero',
+				   '-Wunsafe-loop-optimizations',
+				   '-Wcast-align',
+				   '-Wlogical-op',
+				   '-Wjump-misses-init']
+	elif compiler.get_id() == 'clang'
+		# TODO add clang specific options
+		compiler_cflags = []
```

Comment:
`-Wnull-dereference -Wdouble-promotion -Wshadow -Wpointer-arith -Wwrite-strings -Wstrict-overflow=4 -Wundef` ?

Reply:
I moved them in the common base_cflags

Reply:
Ack